### ISSUE TITLE
Wrong TimeUnit of PrometheusTimer-Total

### ISSUE THREAD

**Issue by MarkusKull** [micrometer-metrics/micrometer#2060]
@jkschneider @shakuzen It seems #2005 #1944 introduced a regression into (standard) PrometheusTimer: The reported Timer-Total (Sum) is now way too small by a scale of 9.
Detected in Micrometer 1.5.0 (SpringBoot 2.3.0.RC1).

PrometheusTimer:122 stores Total as seconds into HistogramSnapshot
```
        return new HistogramSnapshot(snapshot.count(),
                snapshot.total(TimeUnit.SECONDS),
                snapshot.max(TimeUnit.SECONDS),
````

But HistogramSnapshot:61 always treats the given Total as nanos
```
    public double total() {
        return total;
    }

    public double total(TimeUnit unit) {
        return TimeUtils.nanosToUnit(total, unit);
    }
```

Above Sources are unchanged for a long time. The problem is now visible because PrometheusMeterRegistry:391 calls overloaded `total` with explicit TimeUnit. Thus falsely converting nanos to seconds, although Total is already seconds.
```
            samples.add(new Collector.MetricFamilySamples.Sample(
                    conventionName + "_sum", tagKeys, tagValues, histogramSnapshot.total(TimeUnit.SECONDS)));
```

I could provide a MR, but it is unclear for me how a fitting fix should look like. What do you propose?

**Comment by jkschneider** [comment]
@MarkusKull Yikes! Thank you for reporting this so quickly.

In testing, I noticed that this is limited to timers that have percentile histograms turned out, which I suppose is why we didn't catch it earlier.

**Comment by yuryfunikov** [comment]
is it possible to merge PR https://github.com/micrometer-metrics/micrometer/pull/2537 to backport this fix to 1.3.x branch?
